drivers/regulator/Kconfig.pf1550

```diff
@@ -0,0 +1,29 @@
+# Copyright (c) 2024 Arduino SA
+# SPDX-License-Identifier: Apache-2.0
+
+config REGULATOR_PF1550
+	bool "NXP PF1550 PMIC regulator driver"
+	default y
+	depends on DT_HAS_NXP_PF1550_REGULATOR_ENABLED
+	select I2C
+	select MFD
+	help
+	  Enable the NXP PF1550 PMIC regulator driver
+
+if REGULATOR_PF1550
+
+config REGULATOR_NXP_PF1550_COMMON_INIT_PRIORITY
+	int "PF1550 regulator driver init priority (common part)"
+	default 86
+	help
+	  Init priority for the NXP PF1550 regulator driver
+	  (common part). It must be greater than I2C and MFD init priority.
+
+config REGULATOR_NXP_PF1550_INIT_PRIORITY
+	int "PF1550 regulator driver init priority"
+	default 87
+	help
+	  Init priority for the NXP PF1550 regulator driver. It must be
+	  greater than REGULATOR_NXP_PF1550_COMMON_INIT_PRIORITY
```

Comment:
if you use the same priority value on all devices they should all get sequence correctly thanks to the subpriority value, want to try that? may get rid of few symbols, I would keep the MFD one for all of them